robottelo/entities.py

```diff
@@ -394,12 +392,13 @@ def create_missing(self, auth=None):
         * the ``template_kind`` instance attribute is unset.
 
         """
-        super(ConfigTemplate, self).create_missing(auth)
+        super(ConfigTemplate, self).create_missing()
         if (vars(self).get('snippet') is False and
                 'template_kind' not in vars(self)):
             # A server is pre-populated with exactly eight template kinds. We
             # use one of those instead of creating a new one on the fly.
             self.template_kind = TemplateKind(
+                self.server_config_,
```

Comment:
Right. When an entity class is created, you have the option of passing in a server configuration object that differs from whatever is in robottelo's configuration file. What we're doing here is making sure that the `ConfigTemplate` object gives its configuration to the `TemplateKind` object being created. This is useful in the following example situation:

```python
config_templates = [
    ConfigTemplate(ServerConfig.get(label))
    for label
    in ('sat1-profile', 'sat2-profile')
]
config_template_attrs = [
    config_template.create_json()
    for config_template
    in config_templates
]
```

In the above example, each of the two config template objects may exist on a different server. In that case, the `create_missing` method should go ahead and create a dependent template kind object on the appropriate server too.